r-rse/course-design.md

```diff
@@ -0,0 +1,135 @@
+    - Incorporates best practices in package development 
+    (unit testing, continuous integration, version control)
+    - Is designed to involve the community, such as gaining potential contributors
+    (including licensing, code of conduct, contributing guidelines, 
+    well described README)
+    - Has user-friendly (i.e. targetted to a novice to intermediate user)
+    documentation and tutorials on how to use the package (e.g. a nice website)
+    - Is adequately structured to make package management and maintanence easier
+    (e.g. with Issue Templates, Project Boards, Labels, contributing guidelines)
+    
+- **Teaching/learning activities**:
+    - In class portion: 
+        - Some lectures for those material that aren't coding heavily
+        (e.g. for lessons to involve community, being a good community member)
+        - Live coding (with instructor)
+        - Exercises that expand on material covered with instructor that 
+        solve a provided problem, with or without scaffolding
+    - Out of class: 
+        - Group work/project creating an R package based on a set of (given?) problems
+        - Formative assessments like coding with immediate feedback on correct/incorrect
+        (via learnr package? Esp with new update to RStudio in version 1.3)
+        - Reading assignments?
+
+- **Assessment and feedback activities**:
+    - Deadlines for milestones to acheive for project, with instructor feedback on progress
+    - Out-of-class formative assessments based on maybe learnr tutorials?
+    - After completing in-class exercises, go over them as class?
+
+- Key resources to use for this:
+  - R packages: usethis, devtools, drake?, pkgdown, testthat, styler, rmarkdown
+  - Online books: [R Packages](https://r-pkgs.org/) 
+  (we differentiate from this amazing book by structuring everything around 
+  this being a course, with exercises, assessments, (group?) projects, 
+  instructions/comments/notes for instructors, etc)
+
+## Chapter outline
+
+Course chapters and content need to be strongly aligned with learning outcome/goal.
+Here is a draft outline to guide development.
+
+- Syllabus
+    - Overview of course
+    - Why to learn it
+    - What makes this course different
+
+- Folder and file structure (RStudio R Projects)
+    - Filepaths, very basics of the shell (in RStudio)
+    - Making use of the fs package for filesystem management
+    
+- Setting up an R package
+    - Using devtools, usethis for development
+        - e.g. `use_devtools()` and `use_usethis()`.
+    - Developing functions
+    - Making and using datasets
+
+- Function development in a package environment
+    - Dependency management
+    - Function documentation (with roxygen2), part 1
+        - To use `@examples` to help with creating function
+
+- (Idea) Additional features to make use of
+    - RStudio addins, snippets, rmarkdown templates, templates
+    - rstudioapi?
```

Comment:
You mean as possible components of a package, right?  I think of these as things beyond functions that an app can provide. Having a survey of them in one place would be really nice (I can't think of anywhere else that does).   

Would shiny apps fit here too?